dashboard/modules/event/event_head.py

```diff
@@ -54,12 +56,18 @@ def _update_events(event_list):
                 job_id = "global"
             if system_event is False:
                 all_job_events[job_id][event_id] = event
-        # TODO(fyrestone): Limit the event count per job.
+
         for job_id, new_job_events in all_job_events.items():
             job_events = DataSource.events.get(job_id, JobEvents())
             job_events.update(new_job_events)
             DataSource.events[job_id] = job_events
 
+            # Limit the # of events cached if it exceeds the threshold.
+            events = DataSource.events[job_id]
+            if len(events) > MAX_EVENTS_TO_CACHE * 1.1:
+                while len(events) > MAX_EVENTS_TO_CACHE:
+                    events.popitem(last=False)
```

Comment:
Now `events` is an ordered dict, so we technically don't guarantee to delete the oldest events based on the time stamp. We will eventually need to use priority queue, but I don't think it is a very high priority now